inginious/frontend/pages/course_user_settings.py

```diff
@@ -0,0 +1,104 @@
+        """ GET request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        current_user = self.database.users.find_one(
+            {"username": username})
+        course_user_settings = current_user.get("course_settings", {})
+        return self.show_page(courseid, course_user_settings.get(courseid, {}), None)
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        course_user_settings = self._sanitize_content(flask.request.form, courseid)
+        if type(course_user_settings)==str:
+            feedback = ("danger",course_user_settings)
+            current_user = self.database.users.find_one(
+                {"username": username})
+            course_user_settings = current_user.get("course_settings", {})
+            return self.show_page(courseid, course_user_settings.get(courseid, {}), feedback)
+
+        self.database.users.update_one({"username": username}, {"$set": {"course_settings." + courseid: course_user_settings}})
+
+        return self.show_page(courseid, course_user_settings, ("success","Course settings successfully updated."))
+
+    def show_page(self, courseid, course_user_settings, feedback):
+        """
+            Definition of the show page method.
+            :param: courseid: the id of the course.
+            :param: course_user_settings: The dict with the settings values.
+            :param: feedback: a tuple with the type of feedback and the feedback. None if there is no feedback.
+            :return:
+        """
+        try:
+            course = self.course_factory.get_course(courseid)
+            course_user_setting_fields = course.get_course_user_settings()
+            return self.template_helper.render("course_user_settings.html", course=course,
+                                               course_user_setting_fields=course_user_setting_fields,
+                                               course_user_settings=course_user_settings, fieldtypes=FieldTypes,
+                                               feedback=feedback)
+        except Exception:
+            return self.template_helper.render("course_unavailable.html")
+
+    def _is_accessible(self, courseid, username):
+        """
+            Verify if course is accessible and that courseid match an existing course.
+            :param: courseid - The id of the course.
+            :param: username - The username of the logged user.
+            :return: A boolean if the course is accessible for the given user.
+        """
+        try:
+            course = self.course_factory.get_course(courseid)
+            return self.user_manager.course_is_user_registered(course, username)
```

Comment:
```suggestion
            return self.user_manager.course_is_user_registered(course, username) and self.user_manager.course_is_open_to_user(course, lti=False)
```

You can be registered to a closed course.